### PR TITLE
Implemented ScoreVerifier

### PR DESCRIPTION
**Pull request by artem-smotrakov**
Here is a list of updates:

- Added `AbstractVerifier` and `ScoreVerifier` classes
- Added `ScoreVerifierTest`
- Renamed `RatingVerifier.failedVectors()` to `runImpl()`
- `runImpl()` returns a list instead of a set
- Fixed checkstyle warnings in the modified files

### CODE REVIEW

File: src/main/java/com/sap/sgs/phosphor/fosstars/model/qa/AbstractVerifier.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  /**
+   * Check the rating against the test vectors, and throws a {@link VerificationFailedException} if
+   * at least one test vector failed.
+   *
+   * @throws VerificationFailedException If at least one test vector failed.
+   */
+  public final void run() throws VerificationFailedException {
+    List<FailedTestVector> failedVectors = runImpl();
+    for (FailedTestVector vector : failedVectors) {
```
Comment (Sachpat) [inline src/main/java/com/sap/sgs/phosphor/fosstars/model/qa/AbstractVerifier.java]: How about printing the logs in specific verifiers by basically extracting this to a method and then teaking the LOG as a parameter from the specific verifier class?

File: src/main/java/com/sap/sgs/phosphor/fosstars/model/tuning/CMAESWeightsOptimization.java [hunk 2]
```diff
@@ -293,7 +300,7 @@ public double value(double[] parameters) {
     }
 
     /**
-     * @return A number of weights which may be adjusted.
+     * Returns a number of weights which may be adjusted.
```
Comment (Sachpat) [inline src/main/java/com/sap/sgs/phosphor/fosstars/model/tuning/CMAESWeightsOptimization.java]: Returns *the number of weights* maybe? as we are specific about weights here?  :)

Comment (artem-smotrakov) [reply src/main/java/com/sap/sgs/phosphor/fosstars/model/tuning/CMAESWeightsOptimization.java]: I agree.

File: src/main/java/com/sap/sgs/phosphor/fosstars/model/qa/AbstractVerifier.java (hunk 1 above)
Comment (artem-smotrakov) [reply src/main/java/com/sap/sgs/phosphor/fosstars/model/qa/AbstractVerifier.java]: Good point.

**Comment by artem-smotrakov** [comment]
Thanks for the review @Sachpat . I have addressed the comments.

**Pull request merged**